File: 04-Arrays/2D-Arrays/FindKey.cpp
#include <iostream>
#include <vector>
using namespace std;

// 10 10 15 20 25 

bool findKey(int arr[][4],int row ,int col,int key){
    for(int i = 0 ; i < row; i++){
     
        for(int j = 0 ; j < col ; j++){
            if(arr[i][j] == key){
                return true;
            }
        }
        
    }

    return false;

  
    
    
}

int main() {

    // int row = 5;
    // int col = 4;
    int key = 51;

    int arr[5][4] = {{1,2,3,4},{2,3,4,1},{5,6,1,3},{2,4,6,8},{1,9,9,6}};
 
    if(findKey(arr,5,4,key)){
        cout<<"element is present";
    }
    else{
        cout<<"Not present ";
    }


   
    
 

    return 0;
}

File: 05-Array_List-InPython/2D-List-In-Python/FindKey.py
def findKey(alist ,key):

    row = len(alist)
    col = len(alist[0])

    for i in range(row):
        for j in range(col):
            if(key == alist[i][j]):
                return True
    
    return False




def main():
    key = 5
    alist = [[1,2,3,4],[2,3,4,1],[5,6,1,3],[2,4,6,8],[1,9,9,6]]
    print(findKey(alist,key))

if __name__ == "__main__":
    main()